### PR TITLE
Add supports of Pixiv & twipple thumbnails

### PR DESCRIPTION
**Pull request by shuuji3**
- [Pixiv](https://pixiv.net): Most popular Japanese fanart & illustration community like [DeviantArt](https://deviantart.com)
- [twipple](http://twipple.jp): Twitter client and image sharing service provided by BIGLOBE, a famous Japanese ISP

You can test thumbnail preview by searching "p.twpl.jp OR p.twipple.jp" or "#pixiv \<your favorite works>" on TweetDeck.

### CODE REVIEW

File: src/js/util/providers/pixiv.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+          const data = JSON.parse(match[1]);
+          return data;
+        })
+        .then(data => {
+          const thumbnailUrl = $.getSafeURL(data.img.replace('240x480', '600x600'));
+          const imgUrl = $.getSafeURL(data.img.replace('240x480', '1200x1200'));
+          return {
+            type: 'image',
+            thumbnail_url: thumbnailUrl,
+            url: $.getSafeURL(imgUrl),
```
Comment (eramdam) [inline src/js/util/providers/pixiv.js]: You already did `getSafeURL` once, you don't need to do it another time

**Comment by shuuji3** [comment]
To get an image, I used [this script](http://source.pixiv.net/source/embed.js) provided by Pixiv to embed images on the website. However, it seems that the large size image (600x600 and 1200x1200) is shown through `getSafeURL()`. Though only 240x480 size image can be shown without `getSafeURL()`, it's too small to view.

**Comment by eramdam** [comment]
@shuuji3 I guess, but you're still doing `getSafeURL()` over the result of `getSafeURL()`, resulting in a very long and buggy URL. I can fix that after merging the PR but it would be better if you fixed it 😄  It's just a matter of changing

```js
url: $.getSafeURL(imgUrl)
```

to 

```js
url: imgUrl
```

**Comment by shuuji3** [comment]
Yes, I can understand what you'd like to say and we shouldn't nest `getSafeURL`. But without nesting `getSafeURL`, we failed to fetch the large image. I assume there is something to block direct access to the large image and Google proxy can avoid it. I don't know what the proxy do but it works well, so I think we need to nest `getSafeURL` to solve the problem. What do you think?

**Comment by eramdam** [comment]
Hum, ok. I will try to see if I can find another approach later. You might also want to [sync your fork](https://help.github.com/articles/syncing-a-fork/), I updated master with some fixes on the modal templates.

**Comment by shuuji3** [comment]
I'm sorry to bother you. Thank you! 😊

**Pull request merged**